### PR TITLE
move to JaCoCo coverage tool

### PR DESCRIPTION
**Pull request by rkatsyuryna**

### CODE REVIEW

File: core/src/test/java/io/github/tcdl/ChannelManagerTest.java [hunk 1]
```diff
@@ -106,7 +106,7 @@ public void testReceiveMessageInvokesAgentAndEmitsEvent() throws InterruptedExce
             awaitReceiveEvents.countDown();
         });
 
-        assertTrue(awaitReceiveEvents.await(3000, TimeUnit.MILLISECONDS));
+        assertTrue(awaitReceiveEvents.await(4000, TimeUnit.MILLISECONDS));
```
Comment (vso-tc) [inline core/src/test/java/io/github/tcdl/ChannelManagerTest.java:109]: I assume these changes are not related to Jacoco plugin. Please commit them separately

Comment (rkatsyuryna) [inline core/src/test/java/io/github/tcdl/ChannelManagerTest.java:109]: actually they are related to slow Travis sometimes so they are a necessary evil

Comment (vso-tc) [inline core/src/test/java/io/github/tcdl/ChannelManagerTest.java:109]: "sometimes" is the key word here
